firewood/src/v2/api.rs

```diff
@@ -170,6 +171,40 @@ pub trait DbView {
         last_key: Option<K>,
         limit: Option<usize>,
     ) -> Result<Option<RangeProof<Vec<u8>, Vec<u8>>>, Error>;
+
+    /// Obtain a stream over the keys/values of this view, using an optional starting point
+    ///
+    /// # Arguments
+    ///
+    /// * `first_key` - If None, start at the lowest key
+    ///
+    /// # Note
+    ///
+    /// If you always want to start at the beginning, [DbView::iter] is easier to use
+    /// If you always provide a key, [DbView::iter_from] is easier to use
+    ///
+    async fn iter_option<K: KeyType>(
+        &self,
+        first_key: Option<K>,
+    ) -> Result<impl Stream<Item = Result<(Box<[u8]>, Vec<u8>), Error>>, Error>;
+
+    /// Obtain a stream over the keys/values of this view, starting from the beginning
+    fn iter(
+        &self,
+    ) -> impl Future<Output = Result<impl Stream<Item = Result<(Box<[u8]>, Vec<u8>), Error>>, Error>>
```

Comment:
`Stream` should be an associated type, then you wouldn't have to use `#[allow(refining_impl_trait)]`